### PR TITLE
Npeinbot

### LINKED ISSUE #1106: Scenarios do not load.
[Raiden.zip](https://github.com/MegaMek/mekhq/files/2897444/Raiden.zip)

Fill out the fields below with as much pertinent information as possible to help the developers solve the issue.
Search for your issue inside open and closed issues first before reporting so as not to make duplicates.
(Delete the above lines)

### Environment
What version of MekHQ does your issue apply to? 45.3
What operating system are you using?  Windows, Mac, or Linux? Windows
What java version are you using?  If unsure it will state your java version at the top of the mekhqlog.txt file. 1.8.0_201-b09

[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/2897440/mekhqlog.txt)
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/2897441/mekhqlog.txt)

### Description
Scenarios do not load.

### Files
[Raiden30550319.zip](https://github.com/MegaMek/mekhq/files/2897442/Raiden30550319.zip)

### PR DESCRIPTION
**Pull request by VicenteCartas**
Fixes #1106

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/mission/BotForce.java [hunk 1]
```diff
@@ -194,9 +195,7 @@ public void writeToXml(PrintWriter pw1, int indent) {
 
         pw1.println(MekHqXmlUtil.indentStr(indent+1) + "<entities>");
         for (Entity en : entityList) {
-            if (en != null) {
-                pw1.println(AtBScenario.writeEntityWithCrewToXmlString(en, indent + 2, entityList));
-            }
+            pw1.println(AtBScenario.writeEntityWithCrewToXmlString(en, indent + 2, entityList));
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/mission/BotForce.java]: The rest of the PR seems to be about protecting vs NPEs, why remove the null check here?

File: MekHQ/src/mekhq/campaign/mission/BotForce.java [hunk 2]
```diff
@@ -115,7 +116,7 @@ public void setName(String name) {
     }
 
     public void setEntityList(ArrayList<Entity> entityList) {
-        this.entityList = entityList;
+        this.entityList = new ArrayList<>(entityList.stream().filter(x -> x != null).collect(Collectors.toList()));
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/mission/BotForce.java]: Could you add a comment here explaining what's being done? It looks like you're removing all null entities from the passed-in list, but it took me a minute to figure it out.

File: MekHQ/src/mekhq/campaign/mission/BotForce.java (hunk 1 above)
Comment (VicenteCartas) [reply MekHQ/src/mekhq/campaign/mission/BotForce.java]: There should be no reasonable way to get a null here anymore as we are filtering the collections in the constructor and in the setter. If someone adds a null item via the getter, I think that should error and then we figure out why that happened, as that seems an error worth catching.

File: MekHQ/src/mekhq/campaign/mission/BotForce.java (hunk 2 above)
Comment (VicenteCartas) [reply MekHQ/src/mekhq/campaign/mission/BotForce.java]: Sure!

File: MekHQ/src/mekhq/campaign/mission/BotForce.java (hunk 1 above)
Comment (NickAragua) [reply MekHQ/src/mekhq/campaign/mission/BotForce.java]: Perhaps then a better way would be to log the presence of a null entity and then move on. An error here would, I believe, prevent the game from being saved.

Comment (VicenteCartas) [reply MekHQ/src/mekhq/campaign/mission/BotForce.java]: Ok, I'll add error logging here.

**Pull request merged**